operator/pkg/controllers/controlplane/controlplane.go

```diff
@@ -58,7 +57,7 @@ func (c *controlPlane) For() controllers.Object {
 
 // Reconcile will reconcile all the components running on the control plane
 func (c *controlPlane) Reconcile(ctx context.Context, object controllers.Object) (res *reconcile.Result, err error) {
-	for _, resource := range []reconciler.Interface{
+	for _, resource := range []v1alpha1.ReconcileFinalize{
```

Comment:
addons and etcd doesn't really need this interface IIUC.

Reply:
They all have the same logic for Reconcile and Finalize